src/v/coproc/tests/fixtures/coproc_test_fixture.cc

```diff
@@ -114,6 +114,10 @@ ss::future<> coproc_test_fixture::restart() {
     _root_fixture = std::make_unique<redpanda_thread_fixture>(
       std::move(data_dir));
     co_await _root_fixture->wait_for_controller_leadership();
+    co_await _client->stop();
```

Comment:
@BenPope is this cool? The idea is what should occur when client encounters broker not found? This happens during a particular test where the `redpanda_thread_fixture` is destructed, then instantiated again to simulate restart